localstack/config.py

```diff
@@ -1268,43 +1265,63 @@ def populate_config_env_var_names():
+    return GATEWAY_LISTEN[0].port
 
 
 def get_protocol():
     return "https" if USE_SSL else "http"
 
 
-def service_url(service_key, host=None, port=None):
+# TODO: refactor internal codebase to use external_service_url and internal_service_url
+def external_service_url(host=None, port=None, protocol=None) -> str:
+    """Returns a service URL to an external client used outside where LocalStack runs.
+    The configurations LOCALSTACK_HOST and USE_SSL can customize these returned URLs.
+    `host` can be used to overwrite the default for subdomains.
+    """
+    protocol = protocol or get_protocol()
+    host = host or LOCALSTACK_HOST.host
+    port = port or LOCALSTACK_HOST.port
+    return f"{protocol}://{host}:{port}"
+
+
+def internal_service_url(host=None, port=None, protocol=None) -> str:
+    """Returns a service URL for internal use within where LocalStack runs.
+    Cannot be customized through LOCALSTACK_HOST because we assume LocalStack runs on the same host (i.e., localhost).
+    """
+    protocol = protocol or get_protocol()
     host = host or LOCALHOST
-    port = port or service_port(service_key)
-    return f"{get_protocol()}://{host}:{port}"
+    port = port or GATEWAY_LISTEN[0].port
+    return f"{protocol}://{host}:{port}"
 
 
-def external_service_url(service_key, host=None, port=None):
-    host = host or LOCALSTACK_HOST.host
-    port = port or service_port(service_key, external=True)
-    return service_url(service_key, host=host, port=port)
+# TODO: Go over all usages and decide whether it's an internal or external usage
+def service_url(service_key, host=None, port=None):
+    """@deprecated: Use `internal_service_url()` instead.
+    We assume that most usages are internal but really need to check and update each usage accordingly.
+    """
+    return internal_service_url(host=host, port=port)
 
 
-# FIXME: we don't separate http and non-http ports any more,
-#        so this function should be removed
+# TODO: go over all usages and replace depending on internal or external usage
 def get_edge_port_http():
+    """@deprecated: Use `localstack_host().port` for external and `GATEWAY_LISTEN[0].port` for internal use.
+    This function is also not needed anymore because we don't separate between HTTP and HTTP ports anymore since
+    LocalStack listens to both."""
     return GATEWAY_LISTEN[0].port
 
 
+# TODO: Go over all usages and decide whether it's an internal or external usage
 def get_edge_url(localstack_hostname=None, protocol=None):
-    port = get_edge_port_http()
-    protocol = protocol or get_protocol()
-    localstack_hostname = localstack_hostname or LOCALSTACK_HOST.host
-    return "%s://%s:%s" % (protocol, localstack_hostname, port)
+    """@deprecated: Use `internal_service_url()` instead.
+    We assume that most usages are internal but really need to check and update each usage accordingly.
+    """
+    return internal_service_url(host=localstack_hostname, protocol=protocol)
```

Comment:
I have the impression that `external_service_url` might be more suitable here (we had 2 few failing tests indicating usages of external_service_url) but it might be mixed as well 🤷 
* `tests/aws/test_network_configuration.py::[1mTestAppsync::test_default_strategy`
* `tests/aws/services/apigateway/test_http_apis.py::[1mTestHttpApis::test_apigw_v2_http_jwt_authorizer`